src/pages/ContentScripts/features/fast-pr/githubService.ts

```diff
@@ -0,0 +1,282 @@
+  handleMessage('loading', t('status_get_github_token'), key);
+  const githubToken = await getGithubToken();
+  if (!githubToken) {
+    handleMessage('error', t('error_get_github_token'), key);
+    return;
+  }
+  const newBranch = generateBranchName();
+  handleMessage('loading', t('status_check_repo_permission'), key);
+  const permissionResult = await checkRepositoryPermission(originalRepo, githubToken);
+  if (!permissionResult.success && permissionResult.message) {
+    handleMessage('error', permissionResult.message, key);
+    return;
+  }
+  let prRepo = originalRepo;
+  let forkOwner: string | null = null;
+  if (!permissionResult.permission) {
+    handleMessage('loading', t('status_get_or_create_fork'), key);
+    const prRepoResult = await getOrCreateFork(originalRepo, githubToken);
+    if (!prRepoResult.success && prRepoResult.message) {
+      handleMessage('error', prRepoResult.message, key);
+      return;
+    }
+    prRepo = prRepoResult.forkName;
+    forkOwner = prRepo.split('/')[0];
+  }
+  //Get the latest SHA submission for the default branch
+  handleMessage('loading', t('status_get_latest_commit_sha'), key);
+  const baseBranchShaResult = await getBranchLatestCommitSha(originalRepo, branch, githubToken);
+  if (!baseBranchShaResult.success && baseBranchShaResult.message) {
+    handleMessage('error', baseBranchShaResult.message, key);
+    return;
+  }
+  const baseBranchSha = baseBranchShaResult.baseBranchSha;
+  //Create a new branch
+  handleMessage('loading', t('status_create_branch'), key);
+  const branchCreated = await createBranch(newBranch, baseBranchSha, prRepo, githubToken);
+  if (!branchCreated.success && branchCreated.message) {
+    handleMessage('error', branchCreated.message, key);
+    return;
+  }
+  //Retrieve the SHA value of the file
+  handleMessage('loading', t('status_get_file_sha'), key);
+  const fileShaResult = await getFileSha(filePath, newBranch, prRepo, githubToken);
+  if (!fileShaResult.success && fileShaResult.message) {
+    handleMessage('error', fileShaResult.message, key);
+    return;
+  }
+  const fileSha = fileShaResult.fileSha;
+  //Create or update file content
+  handleMessage('loading', t('status_update_file_content'), key);
+  const fileUpdatedResult = await createOrUpdateFileContent(
+    filePath,
+    fileContent,
+    newBranch,
+    fileSha,
+    prRepo,
+    githubToken
+  );
+  if (!fileUpdatedResult.success && fileUpdatedResult.message) {
+    handleMessage('error', fileUpdatedResult.message, key);
+    return;
+  }
+  //Create a new PR
+  handleMessage('loading', t('status_create_pr'), key);
+  const prUrlResult = await createPullRequest(
+    values.title,
+    values.content,
+    newBranch,
+    branch,
+    forkOwner,
+    originalRepo,
+    githubToken
+  );
+  if (!prUrlResult.success && prUrlResult.message) {
+    handleMessage('error', prUrlResult.message, key);
+    return;
+  }
+  handleMessage('success', t('success_create_pr'), key);
```

Comment:
Can we make this message to show longer and provide the PR URL link, so users can directly click the link to the PR?